### PR TITLE
TYP: Add type hints to functions related to sample datasets

### PR DESCRIPTION
**Pull request by seisman**
Just add type hints and wrap/improve docstrings.

**Preview**: 

- https://pygmt-dev--3077.org.readthedocs.build/en/3077/api/generated/pygmt.datasets.list_sample_data.html#pygmt.datasets.list_sample_data
- https://pygmt-dev--3077.org.readthedocs.build/en/3077/api/generated/pygmt.datasets.load_sample_data.html#pygmt.datasets.load_sample_data

### CODE REVIEW

**Review by michaelgrund — APPROVED** [review]

File: pygmt/datasets/samples.py [hunk 1]
```diff
@@ -131,36 +125,32 @@ def _load_hotspots():
...
+    Data and information are from Smith, D. E., and M. T. Zuber (1996), The shape of
+    Mars and the topographic signature of the hemispheric dichotomy.
 
     Returns
     -------
-    data : pandas.DataFrame
-        The data table with column names "longitude", "latitude", and
-        "radius_m".
+    data
+        The data table with column names "longitude", "latitude", and "radius_m".
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:138]: To make the formulations consistent.

```suggestion
        The data table. The column names are "longitude", "latitude", and "radius_m".
```

File: pygmt/datasets/samples.py [hunk 2]
```diff
@@ -131,36 +125,32 @@ def _load_hotspots():
...
-    Loads a table of rock sample compositions.
+    Load a table of rock sample compositions.
 
     Returns
     -------
-    data : pandas.DataFrame
-        The data table with columns "limestone", "water", "air",
-        and "permittivity".
+    data
+        The data table with columns "limestone", "water", "air", and "permittivity".
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:153]:
```suggestion
        The data table. The column names are "limestone", "water", "air", and "permittivity".
```

File: pygmt/datasets/samples.py [hunk 3]
```diff
@@ -171,26 +161,28 @@ def _load_rock_sample_compositions():
...
-    Load Table 5.11 in Davis: Statistics and Data Analysis in Geology.
+    Load a table of Notre Dame topography.
+
+    The data is Table 5.11 in Davis: Statistics and Data Analysis in Geology.
 
     Returns
     -------
-    data : pandas.DataFrame
+    data
         The data table with columns "x", "y", and "z".
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:173]:
```suggestion
        The data table. The column names are "x", "y", and "z".
```

File: pygmt/datasets/samples.py [hunk 4]
```diff
@@ -171,26 +161,28 @@ def _load_rock_sample_compositions():
...
-def _load_maunaloa_co2():
+def _load_maunaloa_co2() -> pd.DataFrame:
     """
     Load a table of CO2 values from Mauna Loa.
 
     Returns
     -------
-    data : pandas.DataFrame
+    data
         The data table with columns "date" and "co2_ppm".
```
Comment (yvonnefroehlich) [inline pygmt/datasets/samples.py:186]:
```suggestion
        The data table. The column names are "date" and "co2_ppm".
```

**Pull request merged**